glue/core/visual.py

```diff
@@ -25,10 +25,11 @@ class VisualAttributes(HasCallbackProperties):
     :param linestyle: The linestyle (``'solid' | 'dashed' | 'dash-dot' | 'dotted' | 'none'``)
     :param marker: The matplotlib marker shape (``'o' | 's' | '^' | etc``)
     :param markersize: The size of the marker (int)
+    :param preferred_cmap: The colormap required (str)
 
     """
 
-    def __init__(self, parent=None, washout=False, color=None, alpha=None, preferred_cmap=None):
+    def __init__(self, parent=None, color=None, alpha=None, preferred_cmap=None):
```

Comment:
Agreeing in principle, but it could break code that was simply passing the args in order – e.g.
`VisualAttributes(parent, False, 'yellow', 0.6)`
will create an instance with invalid `color` and `alpha` attributes after this change.

The `param` listing is slightly misleading in itself since the other parameters are hardcoded in `__init__` and can only be changed by the `setter`s, not on creation. Should probably be clarified as well (or perhaps better, added as `kwargs` accepted by `__init__`?), but that's rather outside the scope of this PR. Also should put this on my list of docstrings yet to be converted to numpydoc format. ;-)

Reply:
> Agreeing in principle, but it could break code that was simply passing the args in order – e.g. `VisualAttributes(parent, False, 'yellow', 0.6)` will create an instance with invalid `color` and `alpha` attributes after this change.

I don't disagree, I don't know a work around to stop this being a breaking change.

> The `param` listing is slightly misleading in itself since the other parameters are hardcoded in `__init__` and can only be changed by the `setter`s, not on creation. Should probably be clarified as well (or perhaps better, added as `kwargs` accepted by `__init__`?), but that's rather outside the scope of this PR. 

I can at least add a ``kwargs`` to the init? 

> Also should put this on my list of docstrings yet to be converted to numpydoc format. ;-)

I can convert it?

Reply:
> I can at least add a `kwargs` to the init?

Adding `linewidth`, `linestyle`, `marker` and `markersize` with the current init values as defaults would make sense IMO.

Reply:
> > Also should put this on my list of docstrings yet to be converted to numpydoc format. ;-)
> 
> I can convert it?

If you want to take this on, you might refer e.g. to https://github.com/glue-viz/glue/pull/2237/files#diff-db5318dc12582d4bf966dfcba587b279c0050d8bdb43a5034719c46e5cffb8b8 for the changes in syntax, thanks!

Reply:
This should all be fixed now.